### PR TITLE
Tour creation - Stop Creation - Tour Update

### PR DESCRIPTION
**Pull request by lhedrick05**
I realize now that I should have broken these up into different branches but I was in too deep already by the time I realized.

### CODE REVIEW

File: src/main/java/liftoff/atlas/getcultured/models/User.java
Comment (Cresence) [inline src/main/java/liftoff/atlas/getcultured/models/User.java:1]: This class has some significant implications of contrasting functionality. 

Everything else seems harmless (I've marked others resolved), this file I'd like to hear the compare/contrast between what was left as template in main vs. what is being used in the Tour creation - branch.

**Review by ghost — CHANGES_REQUESTED** [review]
Insight on User.java needed before merge approval

File: src/main/java/liftoff/atlas/getcultured/models/User.java
Comment (AlexMerchant) [reply src/main/java/liftoff/atlas/getcultured/models/User.java:1]: Noting this for our call:

Major differences in how the User is implemented in this 'tour-creation' branch against my most recent implementation in my 'user-authentication' branch are:

- I do not extend my class from AbstractEntity
- I have the ID of the user's id as 'userID' and it is defined directly in User.java
- I have the username of the user's defined as 'username' and directly in User.java
- tour-creation: userEmail member; user-auth: emailAddress
- passwordHash defined on User and acocmpanying methods
- I handle parameterized @Validations in the user sign-up DTO
- UserGroup associations are defined in a Set<UserGroup> attribute; with a @ManyToMany relationship
- Methods to add and remove UserGroup to the 'userGroups' Set; as well as getter and setters
- Set up a @OneToOne relationship between User and UserProfileDetails
- Moved firstName, lastName, location (and added aboutMe) to UserProfileDetails class
- Moved List<Tour> and List<UserReview> properties to UserProfileDetails class
- Slightly varied constructors
- Other getters and setters

**Review by ghost — APPROVED** [review]
Merge conflicts resolved in group meeting. Looks good to me.

**Pull request merged**
